api/src/main/java/org/apache/iceberg/transforms/Days.java

```diff
@@ -55,14 +56,15 @@ public boolean satisfiesOrderOf(Transform<?, ?> other) {
     }
 
     if (other instanceof Timestamps) {
-      return Timestamps.DAY.satisfiesOrderOf(other);
+      ChronoUnit otherResultTypeUnit = ((Timestamps) other).getResultTypeUnit();
+      return otherResultTypeUnit == ChronoUnit.DAYS
+          || otherResultTypeUnit == ChronoUnit.MONTHS
+          || otherResultTypeUnit == ChronoUnit.YEARS;
```

Comment:
I'd prefer to keep the logic of this check inside of the `satisfiesOrderOf` function:
```suggestion
      ChronoUnit otherTimestamp = ((Timestamps) other).getResultTypeUnit();
      if (otherTimestamp == ChronoUnit.MICROS) {
        return Timestamps.DAY_FROM_MICROS.satisfiesOrderOf(other);
      } else if (otherTimestamp == ChronoUnit.NANOS) {
        return Timestamps.DAY_FROM_NANOS.satisfiesOrderOf(other);
      } else {
        throw new UnsupportedOperationException("Unsupported timestamp unit: " + otherTimestamp);
      }
```
Same goes for the other transforms.

Reply:
Done, except for `Hours.java`, which I noticed wasn't delegating to `Timestamps.HOUR` before our change, and so I have left it not delegating to `Timestamps.HOUR_FROM_MICROS` now. That might be something to look into for a future pull request.